GW2EIEvtcParser/EncounterLogic/Raids/W3/KeepConstruct.cs

```diff
@@ -445,5 +446,32 @@ internal override void ComputePlayerCombatReplayActors(AbstractPlayer p, ParsedE
                 }
             }
         }
+
+        protected override void SetInstanceBuffs(ParsedEvtcLog log)
+        {
+            base.SetInstanceBuffs(log);
+            IReadOnlyList<AbstractBuffEvent> downDownDowned = log.CombatData.GetBuffData(AchievementEligibilityDownDownDowned);
+            bool hasBeenAdded = false;
+
+            if (log.FightData.Success)
+            {
+                if (downDownDowned.Any())
+                {
+                    foreach (Player p in log.PlayerList)
+                    {
+                        if (p.HasBuff(log, AchievementEligibilityDownDownDowned, log.FightData.FightEnd - ServerDelayConstant))
+                        {
+                            InstanceBuffs.Add((log.Buffs.BuffsByIds[AchievementEligibilityDownDownDowned], 1));
+                            hasBeenAdded = true;
+                            break;
+                        }
+                    }
+                }
+                if (GetEncounterMode(log.CombatData, log.AgentData, log.FightData) == FightData.EncounterMode.CM && !hasBeenAdded)
+                {
+                    InstanceBuffs.Add((log.Buffs.BuffsByIds[AchievementEligibilityDownDownDowned], 1));
```

Comment:
since we're checking for cm and that all players have hit kc, wouldn't it make more sense to scrap the eligibility check since it's no longer needed?

![5021](https://user-images.githubusercontent.com/23245218/229290235-3049f49c-6af3-499b-a736-14f598496964.png)

Reply:
You can access KC actor immediately via Targets.FirstOrDefault(x => x.IsSpecies(KeepConstruct)); but yes, would simplify stuff quite a bit, you should also check for CM in the first if.